app/parse.py

```diff
@@ -15,8 +31,126 @@ class Product:
     num_of_reviews: int
 
 
+def init_driver() -> webdriver.Chrome:
+    options = Options()
+    options.add_argument("--headless")
+    options.add_argument("--no-sandbox")
+    options.add_argument("--disable-dev-shm-usage")
+
+    driver = webdriver.Chrome(
+        service=ChromeService(ChromeDriverManager().install()), options=options
+    )
+    return driver
+
+
+def accept_cookies(driver: webdriver.Chrome) -> None:
+    try:
+        accept_button = WebDriverWait(driver, 2).until(
+            ec.element_to_be_clickable((By.CLASS_NAME, "acceptCookies"))
+        )
+        accept_button.click()
+        print("Cookies accepted")
+    except Exception as e:
+        print(f"Cookies not accepted: {e}")
+
+
+def load_all_products(driver: webdriver.Chrome) -> None:
+    while True:
+        try:
+            more_button = WebDriverWait(driver, 2).until(
+                ec.presence_of_element_located(
+                    (By.CLASS_NAME, "ecomerce-items-scroll-more")
+                )
+            )
+
+            if (
+                "disabled" not in more_button.get_attribute("class")
+                and more_button.is_displayed()
+            ):
+                more_button.click()
+                print("Clicked 'More' button")
+            else:
+                print("'More' button is either disabled or not displayed")
+                break
+        except Exception as e:
+            print("The 'More' button was not found or became unavailable:", e)
+            break
+
+
+def scrape_page_products(driver: webdriver.Chrome) -> List[Product]:
+    products = []
+    product_elements = driver.find_elements(By.CLASS_NAME, "thumbnail")
+
+    for product in product_elements:
+        try:
+            title = (
+                product.find_element(By.CLASS_NAME, "title")
+                .get_attribute("title")
+                .strip()
+            )
+            description = product.find_element(
+                By.CLASS_NAME, "description"
+            ).text.strip()
+            price = float(
+                product.find_element(By.CLASS_NAME, "price")
+                .text.strip()
+                .replace("$", "")
+            )
+            rating = len(product.find_elements(By.CLASS_NAME, "ws-icon-star"))
+            num_of_reviews = int(
+                product.find_element(By.CLASS_NAME, "ratings").text.split()[0]
+            )
+
+            products.append(Product(
+                title,
+                description,
+                price, rating,
+                num_of_reviews)
+            )
+        except Exception as e:
+            print(f"Error scraping product: {e}")
```

Comment:
Again, catching a general `Exception` is not ideal. Consider handling specific exceptions that are likely to occur during the scraping process, such as `NoSuchElementException` or `ValueError` for conversion issues.